src/components/MapRoute/index.tsx

```diff
@@ -0,0 +1,18 @@
+import {RideModel, HistoryModel, Q, db} from '../../database';
+import withObservables from '@nozbe/with-observables';
+import GeoJSONRoute from './GeoJSONRoute';
+import {historyToGeoJSON} from '../../lib/data';
+
+const RideRoute = ({data}: {data: HistoryModel[]}) => {
+  const geojson = historyToGeoJSON(data);
+  return <GeoJSONRoute geojson={geojson} />;
+};
+
+const enhance = withObservables([], ({rideId}: {rideId: RideModel['id']}) => ({
+  data: db
+    .get<HistoryModel>('history')
```

Comment:
Not sure it will work with observables but to get a downsampled history set we could do a raw query like:
```
WITH downsampled_data AS (
  SELECT
    strftime('%Y-%m-%d %H:%M:00', timestamp) AS minute,
    MIN(timestamp) AS min_timestamp
  FROM
    history
  GROUP BY
    minute
)
SELECT
  h.timestamp,
  h.latitude,
  h.longitude
FROM
  history h
JOIN
  downsampled_data dd ON h.timestamp = dd.min_timestamp
ORDER BY
  h.timestamp;
```

Haven't tested if all of that is doable in android SQLite but worth a try. 

One other thing to maybe look at is if there is a query that grabs only unique values for lat and long. Since we don't really want to render gps points where we are basically standing still.